src/twisted/trial/_dist/test/test_workertrial.py

```diff
@@ -40,16 +57,57 @@ def test_emit(self):
         L{WorkerLogObserver} forwards data to L{managercommands.TestWrite}.
         """
         calls = []
-
-        class FakeClient:
-            def callRemote(self, method, **kwargs):
-                calls.append((method, kwargs))
-
-        observer = WorkerLogObserver(FakeClient())
+        observer = WorkerLogObserver(AMPSpyClient(calls))
         observer.emit({"message": ["Some log"]})
         self.assertEqual(calls, [(managercommands.TestWrite, {"out": "Some log"})])
 
 
+class WorkerStdoutTests(TestCase):
+    """
+    Tests for L{WorkerStdout}.
+    """
+
+    def test_write(self):
+        """
+        L{WorkerStdout} forwards data to L{managercommands.TestWrite} without
+        any extra processing.
+        """
+        calls = []
+        stdout = WorkerStdout(AMPSpyClient(calls))
+
+        stdout.write("Her comes the \N{sun}!")
+
+        self.assertEqual(
+            calls, [(managercommands.TestWrite, {"out": "Her comes the \N{sun}!"})]
+        )
+
+    def test_write_bytes(self):
+        """
+        Raises TypeError when a non string value is written.
+        """
+        calls = []
+        stdout = WorkerStdout(AMPSpyClient(calls))
+
+        error = self.assertRaises(TypeError, stdout.write, b"\xe2\x99")
+
+        self.assertEqual("string argument expected, got <class 'type'>", error.args[0])
+
+    def test_integration(self):
+        """
+        This test is here to trigger a write to sys.stdout for which
+        when running in distributed trial workers the value is forwarded to
+        the centralized log.
+
+        It's kind of a manual test, as there is no automatic assertion.
+        When call as
+        C{trial -j1 twisted.trial._dist.test.test_workertrial.WorkerStdoutTests.test_integration}
+
+        You should see the message via C{cat _trial_temp/0/test.log}
+        """
+        sys.stdout.write("Hello from \N{sun}y worker!")
+        sys.stdout.flush()
```

Comment:
I have no idea how to to test this in other way.

There is a unit test for `main()` but I also want to have a test in which the real code is exercised.